apps/decap/src/collections/page.ts

```diff
@@ -166,64 +171,101 @@ const BlockMediaImageSchema = z
-  fs.readdirSync(dir)
-    .filter((file) => file.endsWith('.yml'))
-    .forEach((file) => {
-      const content = yaml.parse(fs.readFileSync(`${dir}/${file}`, 'utf-8'));
-      const id = Object.values(content)
-        .map((page: any) => page.id)
-        .filter((id) => !!id)
-        .pop();
-      Object.keys(content).forEach((lang) => {
-        if (Object.keys(content[lang]).length < 2) {
-          return;
-        }
-        const input = {
-          ...content[lang],
-          id,
-          locale: lang,
-        };
-        const page = pageSchema.safeParse(input);
-        if (page.success) {
-          pages.push([
-            `${page.data.id}:${lang}`,
-            { ...page.data, _decap_id: id },
-          ]);
-        } else {
-          console.warn(`Error parsing ${file} (${lang}):`);
-          console.warn(page.error.message);
-          console.warn('Input:', content[lang]);
-        }
+  if (pages.length === 0) {
+    fs.readdirSync(dir)
+      .filter((file) => file.endsWith('.yml'))
+      .forEach((file) => {
+        const content = yaml.parse(fs.readFileSync(`${dir}/${file}`, 'utf-8'));
+        const id = Object.values(content)
+          .map((page: any) => page.id)
+          .filter((id) => !!id)
+          .pop();
+        const translations: Array<SourcePage> = [];
+        Object.keys(content).forEach((lang) => {
+          if (Object.keys(content[lang]).length < 2) {
+            return;
+          }
+          const input = {
+            ...content[lang],
+            id,
+            locale: lang,
+          };
+          const page = pageSchema.safeParse(input);
+          if (page.success) {
+            translations.push(page.data);
+          } else {
+            console.warn(`Error parsing ${file} (${lang}):`);
+            console.warn(page.error.message);
+            console.warn('Input:', content[lang]);
+          }
+        });
+        const original = { ...translations[0] };
+        original.translations = translations;
+        pages.push(original);
       });
-    });
+  }
   return pages;
 };
+
+export const pageResolvers = (path: string) =>
+  ({
+    SSGPagesResult: {
+      rows: () => getPages(path),
+      total: () => 0,
+    },
+    Query: {
+      ssgPages: () => {
+        return {
+          __typename: 'SSGPagesResult',
+          total: 0,
+          rows: [],
+        };
+      },
+      viewPage: (_, { path }) => {
+        const pages = getPages(path);
+        let result: SourcePage | undefined;
+        for (const page of pages) {
+          if (page.translations) {
+            for (const translation of page.translations) {
+              if (translation?.path === path) {
+                result = { ...translation };
+                result.translations = page.translations;
+              }
+            }
+          }
+        }
+        return result;
+      },
+    },
+  }) satisfies SourceResolvers;
```

Comment:
Yes, satisfies makes sure the type matches `SourceResolvers` but if the actual type is more specific, it will not loose that information, while a return type definition does.